### ISSUE TITLE
eksctl cluster create fails with 'missing CertificateAuthorityData'

### ISSUE THREAD

**Issue by theimpostor** [weaveworks/eksctl#882]
**What happened?**
The basic eksctl create cluster command failed:
```
> eksctl create cluster
...
[✖]  invalid cluster config: missing CertificateAuthorityData
```

Adding the --zones parameter as suggested by https://linuxacademy.com/community/posts/show/topic/37139-eks-cluster-creation-errorinvalid-cluster-config works:
```
> eksctl create cluster --region=us-west-2 --zones=us-west-2b,us-west-2c,us-west-2d
...
[✔]  EKS cluster "attractive-gopher-1560447986" in "us-west-2" region is ready
```

**Anything else we need to know?**
What OS are you using, are you using a downloaded binary or did you compile eksctl, what type of AWS credentials are you using (i.e. default/named profile, MFA) - please don't include actual credentials though!
macos high sierra, using eksctl via homebrew weaveworks tap.

**Versions**
Please paste in the output of these commands:
```
$ eksctl version
[ℹ]  version.Info{BuiltAt:"", GitCommit:"", GitTag:"0.1.35"}
$ uname -a
Darwin Sahirs-MacBook-Pro.local 17.7.0 Darwin Kernel Version 17.7.0: Wed Apr 24 21:17:24 PDT 2019; root:xnu-4570.71.45~1/RELEASE_X86_64 x86_64
$ kubectl version
Client Version: version.Info{Major:"1", Minor:"14", GitVersion:"v1.14.3", GitCommit:"5e53fd6bc17c0dec8434817e69b04a25d8ae0ff0", GitTreeState:"clean", BuildDate:"2019-06-07T09:57:54Z", GoVersion:"go1.12.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"12+", GitVersion:"v1.12.6-eks-d69f1b", GitCommit:"d69f1bf3669bf00b7f4a758e978e0e7a1e3a68f7", GitTreeState:"clean", BuildDate:"2019-02-28T20:26:10Z", GoVersion:"go1.10.8", Compiler:"gc", Platform:"linux/amd64"}

```

**Logs**
Include the output of the command line when running eksctl. If possible, eksctl should be run with debug logs. For example:
`eksctl get clusters -v 4`
Make sure you redact any sensitive information before posting.
If the output is long, please consider a Gist.
https://gist.github.com/theimpostor/80741a64fbb76d16ac74e7479439c689

**Comment by npockrus-bnet** [comment]
Just to add to this, if you are trying to deploy using a yaml file with the lines below, the cluster still fails to create with the same error. It seems you cannot create a cluster from yaml deploys.

`availabilityzones:
  - zone1
  - zone2
  ...
`

**Comment by martina-if** [comment]
Hi @theimpostor and @npockrus-bnet , we found the bug that was causing this issue. It will be addressed in the next release. Thank you for the report!

**Comment by errordeveloper** [comment]
Yes, `invalid cluster config: missing CertificateAuthorityData` error is a misleading one (it was fixed in #866). We have also reduced the verbosity of the output, so it should be easier to see the root cause (#865).

The issue with zones is a know EKS issue, we don't have a way of addressing it directly in eksctl, but we certainly would like to improve the output messages (see #491).

Given the above, I'm going to close this.
